Source/WebKitLegacy/mac/Misc/WebNSDataExtras.mm

```diff
@@ -56,128 +59,118 @@ - (NSString *)_webkit_guessedMIMETypeForXML
         return MIMEType;
 
-    NSUInteger length = [self length];
-    const char* bytes = static_cast<const char*>([self bytes]);
+    auto bytes = span(self);
 
-    const char* p = bytes;
-    int remaining = std::min<NSUInteger>(length, WEB_GUESS_MIME_TYPE_PEEK_LENGTH) - (SCRIPT_TAG_LENGTH - 1);
-    while (remaining > 0) {
+    size_t remaining = std::min<size_t>(bytes.size(), WEB_GUESS_MIME_TYPE_PEEK_LENGTH) - (scriptTagLength - 1);
+    auto cursor = bytes.first(remaining);
+    while (!cursor.empty()) {
         // Look for a "<".
-        const char* hit = static_cast<const char*>(memchr(p, '<', remaining));
-        if (!hit)
+        size_t hitIndex = WTF::find(cursor, '<');
+        if (hitIndex == notFound)
             break;
 
+        auto hit = cursor.subspan(hitIndex);
         // If we found a "<", look for "<html>" or "<a " or "<script".
-        if (!strncasecmp(hit, "<html>", strlen("<html>"))
-            || !strncasecmp(hit, "<a ", strlen("<a "))
-            || !strncasecmp(hit, "<script", strlen("<script"))
-            || !strncasecmp(hit, "<title>", strlen("<title>"))) {
+        if (spanHasPrefixIgnoringASCIICase(hit, "<html>"_span)
+            || spanHasPrefixIgnoringASCIICase(hit, "<a "_span)
+            || spanHasPrefixIgnoringASCIICase(hit, "<script"_span)
+            || spanHasPrefixIgnoringASCIICase(hit, "<title>"_span)) {
             return @"text/html";
         }
 
         // Skip the "<" and continue.
-        remaining -= (hit + 1) - p;
-        p = hit + 1;
+        skip(cursor, hitIndex + 1);
     }
 
     // Test for a broken server which has sent the content type as part of the content.
     // This code could be improved to look for other mime types.
-    p = bytes;
-    remaining = std::min<NSUInteger>(length, WEB_GUESS_MIME_TYPE_PEEK_LENGTH) - (TEXT_HTML_LENGTH - 1);
-    while (remaining > 0) {
+    remaining = std::min<size_t>(bytes.size(), WEB_GUESS_MIME_TYPE_PEEK_LENGTH) - (textHTMLLength - 1);
+    cursor = bytes.first(remaining);
+    while (!cursor.empty()) {
         // Look for a "t" or "T".
-        const char* hit = nullptr;
-        const char* lowerhit = static_cast<const char*>(memchr(p, 't', remaining));
-        const char* upperhit = static_cast<const char*>(memchr(p, 'T', remaining));
-        if (!lowerhit && !upperhit)
+        size_t lowerHitIndex = WTF::find(cursor, 't');
+        size_t upperHitIndex = WTF::find(cursor, 'T');
+        if (lowerHitIndex == notFound && upperHitIndex == notFound)
             break;
 
-        if (!lowerhit)
-            hit = upperhit;
-        else if (!upperhit)
-            hit = lowerhit;
+        size_t hitIndex;
+        if (lowerHitIndex == notFound)
+            hitIndex = upperHitIndex;
+        else if (upperHitIndex == notFound)
+            hitIndex = lowerHitIndex;
         else
-            hit = std::min<const char*>(lowerhit, upperhit);
+            hitIndex = std::min<size_t>(lowerHitIndex, upperHitIndex);
```

Comment:
If `notFound` is higher than any other `size_t` value then this whole thing can just be:
```
static_assert(notFound == std::numeric_limits<size_t>::max());
size_t hitIndex = std::min(lowerHitIndex, upperHitIndex);
```
No need to write `std::min<size_t>`.

Reply:
Good idea. Will update.